collections/list.go

```diff
@@ -89,6 +104,25 @@ func (l *List[T]) IndexOf(item T) int {
 	return -1
 }
 
+// Use Map with a callback function to transform given list to a different one.
+// Map has a return type od ListTransformation and the returned value should be passed to the CopyFrom method to generate the required list.
+func (l *List[T]) Map(callback listMapFunction) ListTransformation {
```

Comment:
```suggestion
func (l *List[T]) Map(callback listMapFunction[T]) ListTransformation {
```
I feel `listMapFunction` can be designed similar to `listReduceFunction`, so that we can avoid casting.

Reply:
@aswath-s-tw Great suggestion! Made the change.